lib/pacemaker/pcmk_output.c

```diff
@@ -1864,6 +1864,65 @@ attribute_xml(pcmk__output_t *out, va_list args)
     return pcmk_rc_ok;
 }
 
+PCMK__OUTPUT_ARGS("rule-check", "const char *", "int", "const char *")
+static int
+rule_check_default(pcmk__output_t *out, va_list args)
+{
+    const char *rule_id = va_arg(args, const char *);
+    int result = va_arg(args, int);
+    const char *error = va_arg(args, const char *);
+
+    switch (result) {
+        case pcmk_rc_within_range:
+            return out->info(out, "Rule %s is still in effect", rule_id);
+        case pcmk_rc_ok:
+            return out->info(out, "Rule %s satisfies conditions", rule_id);
+        case pcmk_rc_after_range:
+            return out->info(out, "Rule %s is expired", rule_id);
+        case pcmk_rc_before_range:
+            return out->info(out, "Rule %s has not yet taken effect", rule_id);
+        case pcmk_rc_op_unsatisfied:
+            return out->info(out, "Rule %s does not satisfy conditions",
+                             rule_id);
+        default:
+            out->err(out,
+                     "Could not determine whether rule %s is in effect: %s",
+                     rule_id, ((error != NULL)? error : "unexpected error"));
+            return pcmk_rc_ok;
+    }
+}
+
+PCMK__OUTPUT_ARGS("rule-check", "const char *", "int", "const char *")
+static int
+rule_check_xml(pcmk__output_t *out, va_list args)
+{
+    const char *rule_id = va_arg(args, const char *);
+    int result = va_arg(args, int);
+    const char *error = va_arg(args, const char *);
+
+    char *rc_str = pcmk__itoa(pcmk_rc2exitc(result));
```

Comment:
I wonder if we should map both success codes (`pcmk_rc_ok` and `pcmk_rc_within_range`) to `pcmk_rc_ok`, to make things simpler for the caller. An rc of 0 would mean "the rule is in effect" -- instead of checking for two different codes. I don't think it would deprive anyone of useful information.

Not sure if we really want `pcmk_rc2exitc` here but I'm leaving it alone unless requested.